smart-contract/deploy/upgrades/upgrade-gatewaystaking.ts

```diff
@@ -0,0 +1,45 @@
+import { addContractToAdmin, getDeploymentSigner, sleep, verify } from ".././defender-utils";
+import { ethers , upgrades } from 'hardhat';
+import { Signer } from '@ethersproject/abstract-signer/src.ts'
+import { BNB_TESTNET_CONTRACT_ADDRESSES } from ".././utils";
+
+async function main() {
+    const testnetTokenContractAddress = BNB_TESTNET_CONTRACT_ADDRESSES.erc20;
+
+    const args = [testnetTokenContractAddress , 'Identity Test Staking Vault', "ID_TEST_STAKE"];
+    
+    const signer: Signer = await getDeploymentSigner();
+
+    const GatewayStakingContractFactory = await ethers.getContractFactory("GatewayStaking", signer!);
+
+    const gatewayStakingContract = await upgrades.upgradeProxy(
+        BNB_TESTNET_CONTRACT_ADDRESSES.gatewayStaking, 
+        GatewayStakingContractFactory, 
+        { 
+            kind: 'uups', 
+            constructorArgs: args,
+            unsafeAllow: ['state-variable-immutable', 'constructor']
```

Comment:
Why do we need to allow "constructor" here?

Reply:
This is the contructor for the gatewayStaking contract:

```solidity
constructor(ERC20 asset_, string memory name_, string memory symbol_) ERC4626(asset_) ERC20(name_, symbol_) {
      _superAdmins[msg.sender] = true;
   }
```

We need to allow the constructor so the staking contract correctly sets up the ERC4626 vault. The alternative is to use upgradable versions of ERC4626 and ERC20 but that would make the initializer logic a bit more complex.

Reply:
I think because this  contract inherits from ERC20 and ERC-4626 adding a comment to allow unsafe isn't working because I can't add the inline comments to the openzepplin ERC20 and ERC-4626 contracts. 
<img width="1030" alt="image" src="https://github.com/identity-com/gateway-protocol-evm/assets/40375385/e62e8a26-51df-4eb7-ad1f-70c00adb0b94">

Reply:
Cool, then this makes sense!